frontend/src/Components/ShoppingCart.tsx

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import APIService from "../APIService"
+import {useState, useEffect} from "react";
+import { 
+    List, 
+    Divider, 
+    ListItem, 
+    Button,
+    ListItemAvatar,
+    ListItemText,
+    Typography,
+    Avatar,
+    Container,
+    ListItemSecondaryAction,
+    IconButton,
+    Paper,
+    CssBaseline
+} from '@material-ui/core';
+
+import DeleteIcon from '@material-ui/icons/Delete';
+import "../Styles/ShoppingCart.css"
+import Product from "../Interfaces/Product";
+import Error from "./Error";
+import Loading from "./Loading";
+
+export default function ShoppingCart(){
+
+    const [cart, setCart] = useState<number[]>([])
+    const [isLoaded, setIsLoaded] = useState<boolean>(false)
+    const [error, setError] = useState<string>()
+    const [items, setItems] = useState<Product[]>([])
+    const [updated, setUpdated] = useState<number>(0)
+
+    useEffect(() => {
+        APIService.Products.list()
+        .then(result => setItems(result))
+        .catch(err => setError(err))
+
+        const storageCart = localStorage.getItem("cart")
+        if(storageCart !== null && storageCart !== ""){
+            setCart(JSON.parse(storageCart))
+        }
+
+        setIsLoaded(true)
+    }, [updated])
+
+    function deleteProductFromCart(id: number) {
+        if(cart.length === 0) return;
+        const index = cart.indexOf(id)
+        if( index === -1 ) return
+        let copyOfArr = [...cart]
+        copyOfArr.splice(index, 1);
+        localStorage.setItem("cart", JSON.stringify(copyOfArr))
+        setCart(copyOfArr)
+        setUpdated(updated + 1)
+    }
+
+    function deleteAllProductFromCart(){
+        localStorage.setItem("cart", "")
+        setCart([])
+    }
+
+    let product: Product;
+    let subTotal = 0;
+
+    return !isLoaded ? (
```

Comment:
1. the error can go into the Container below

2. create a reusable loading component, pass in the loading state and let it handle the logic

3. too many ternary operators will make the code less unreadable. Put the empty shopping cart and a non-empty one within the same container will reduce the number of IFs